src/cargo/core/shell.rs

```diff
@@ -377,7 +377,7 @@ mod imp {
     pub fn stderr_width() -> Option<usize> {
         unsafe {
             let mut winsize: libc::winsize = mem::zeroed();
-            if libc::ioctl(libc::STDERR_FILENO, libc::TIOCGWINSZ.into(), &mut winsize) < 0 {
+            if libc::ioctl(libc::STDERR_FILENO, libc::TIOCGWINSZ, &mut winsize) < 0 {
```

Comment:
This was changed specifically for freebsd, I don't think it should be modified.

Reply:
Alright, I backed that one out.

Reply:
Can you add `#[allow(clippy::identity_conversion)]` with a comment explaining why?

Reply:
[done](https://github.com/rust-lang/cargo/pull/7251/files#diff-b53e068c2a295bd3080db1f2a6491fcfR378)

Reply:
FWIW I'm personally not a fan of adding `#[allow(clippy::...)]` throughout the code, I find it pretty jarring to basically see a random distribution of clippy lints sprayed throughout the code. I think it's pretty bad style and is one of the reasons I'm not a fan of clippy, I'd rather not have a tool force me to randomly annotate code where it just otherwise causes bugs.

Reply:
Would it be OK to add it to `src/cargo/lib.rs` with the rest of the list?

I agree there are too many false positives in clippy's default set, and that it can be too pedantic. However, I think there are enough cases where it does help simplify and clarify code, and even point out invalid code.  For example, it points out a memory corruption issue in git2 that I've been meaning to fix, but haven't got around to.  

I also think there should have been a comment here in the first place.  A magical `.into` that only applies to one target that isn't even tested on CI should have *something* to explain why it is there.

Reply:
Yes I think it'd at least be better to go in `src/cargo/lib.rs`. I also do understand that this is a pretty unproductive location to litigate this, it's not like anyone from clippy is watching this PR. I apologize for venting here :(

Reply:
I suppressed the warning in `lib.rs` and reverted the change to `shell.rs`.